draft-ietf-quic-multipath.md

```diff
@@ -522,31 +514,46 @@ for subsequent connections, it might be difficult for the client
 to pick the "right" token among multiple tokens obtained in a previous connection.
 The client is likely to fall back to the strategy specified in {{Section 8.1.3 of QUIC-TRANSPORT}},
 i.e., pick the last received token. To avoid issues when clients make the "wrong" choice,
-a server should issue a token that is capable of validating
+a server SHOULD issue tokens that are capable of validating
 any of the previously validated addresses. Further guidance on token usage can be
 found in {{Section 8.1.3 of QUIC-TRANSPORT}}.
 
 ## Handling Connection IDs {#consume-retire-cid}
 
-With the multipath extension, each connection ID is associated with one path
-that is identified by the Path ID that is specified in the Path Identifier field of
-the PATH_NEW_CONNECTION_ID frame {{mp-new-conn-id-frame}}.
-The Path ID 0 indicates the initial path of the connection.
-Respectively, the connection IDs used during the handshake belong to the initial path
-with Path ID 0.
+In order to identify a connection ID correctly when the multipath extension is used,
+endpoints have to use the PATH_NEW_CONNECTION_ID and PATH_RETIRE_CONNECTION_ID frames
+instead of the NEW_CONNECTION_ID and RETIRE_CONNECTION_ID frame to indicate
+the respective Path ID together with associated sequence number
+(see {{Section 5.1.1 of QUIC-TRANSPORT}}), at least for all paths with a Path ID other than 0.
+Each Path ID has its own connection ID sequence number space whose initial value is 0.
+
+Endpoints can also use PATH_NEW_CONNECTION_ID and
+PATH_RETIRE_CONNECTION_ID for the initial path with Path ID 0,
+however, the use of NEW_CONNECTION_ID and RETIRE_CONNECTION_ID
+is still valid and endpoints need to process these frames accordingly
+as corresponding to Path ID 0.
 
 ### Issuing New Connection IDs
 
-The PATH_NEW_CONNECTION_ID frame is used to issue new connection IDs for all paths.
+On non-initial paths
+(i.e., Path ID different from 0), the initial path connection ID is provided by a
+PATH_NEW_CONNECTION_ID frame with a sequence number value of 0.
 In order to let the peer open new paths, it is RECOMMENDED to proactively
```

Comment:
No. Previous text is better. You may want to say that it is OK to send legacy NEW_CONNECTION_ID on path 0, but your text ends up saying that NEW_CONNECTION_ID frames cannot be used on path with number not equal to zero,
which is not what we agreed. How about:

The PATH_NEW_CONNECTION_ID frame is used to issue new connection IDs for any path. The NEW_CONNECTION_ID frame MAY be used to provide new connection IDs for path number 0. Connection ID sequence numbers start at 0.

Reply:
This is not a meant as a replacement. I simply moved this sentence up (see removed lines 547, 548, and 579 below to find the original sentence). And I removed the first sentence because I found it redundant with the text above (see lines 523-527) and also text in the frames section.

However, we can maybe also just remove this sentence as well as we say in line 528 already this:
"Each Path ID has its own connection ID sequence number space whose initial value is 0."

Reply:
OK, let's remove the sentence.

Reply:
done